gravitee-apim-rest-api/gravitee-apim-rest-api-management/gravitee-apim-rest-api-management-rest/src/main/java/io/gravitee/rest/api/management/rest/resource/v4/api/ApiPlansResource.java

```diff
@@ -159,7 +162,8 @@ public Response createApiPlan(@Parameter(name = "plan", required = true) @Valid
         newPlanEntity.setApiId(apiId);
         newPlanEntity.setType(PlanType.API);
 
-        PlanEntity planEntity = planService.create(GraviteeContext.getExecutionContext(), newPlanEntity);
+        final var api = apiSearchService.findById(GraviteeContext.getExecutionContext(), apiId);
```

Comment:
You don't know what the api service does or will do, it may involve other repository calls to build the API, and it seems here you don't even use the API to call extra service yourself.